### PR TITLE
core: Random fixes around ardupilot_manager

### PR DESCRIPTION
**Pull request by patrickelectric**

### CODE REVIEW

**Review by rafaellehmkuhl — CHANGES_REQUESTED** [review]
Python's map require a callable for the first argument. It can be a regular function or a lambda.
I don't think, though, that the lambda version get's any better than the list comprehension. If fact it gets much less readable, while the regular function adds 3 lines of code.

**Comment by patrickelectric** [comment]
> Python's map require a callable for the first argument. It can be a regular function or a lambda.
> I don't think, though, that the lambda version get's any better than the list comprehension. If fact it gets much less readable, while the regular function adds 3 lines of code

```
class A:
    def a(self):
        return "potato"
        
a_list = [A(), A()]

print(list(map(A.a, a_list)))
```

**Comment by Williangalvani** [comment]
I honestly prefer reading list comprehensions

File: core/services/ardupilot_manager/mavlink_proxy/Endpoint.py [hunk 1]
```diff
@@ -57,7 +57,7 @@ def is_mavlink_endpoint(cls: Type["Endpoint"], values: Any) -> Any:
     def __str__(self) -> str:
         return ":".join([self.connection_type, self.place, str(self.argument)])
 
-    def asdict(self) -> Dict[str, Any]:
+    def as_dict(self) -> Dict[str, Any]:
```
[the remark this answers is not in the document]
Comment (patrickelectric) [reply core/services/ardupilot_manager/mavlink_proxy/Endpoint.py:60]: I was unable to find such function in pydantic.

File: core/services/ardupilot_manager/ArduPilotManager.py [hunk 2]
```diff
@@ -162,7 +162,7 @@ def _get_configuration_endpoints(self) -> Set[Endpoint]:
         return {Endpoint(**endpoint) for endpoint in self.configuration.get("endpoints") or []}
 
     def _save_endpoints_to_configuration(self, endpoints: Set[Endpoint]) -> None:
-        self.configuration["endpoints"] = [endpoint.asdict() for endpoint in endpoints]
+        self.configuration["endpoints"] = list(map(Endpoint.asdict, endpoints))
```
[the remark this answers is not in the document]
Comment (rafaellehmkuhl) [reply core/services/ardupilot_manager/ArduPilotManager.py]: Ok, just realized you're not calling the function. My mistake.

**Review by rafaellehmkuhl — APPROVED** [review]

**Comment by rafaellehmkuhl** [comment]
Curiosity: how did you fix the mypy complaints?

**Comment by patrickelectric** [comment]
> Curiosity: how did you fix the mypy complaints?

as @Williangalvani suggestion, it appears that a non static function is necessary to iterate over a inheritance class.

**Review by Williangalvani — APPROVED** [review]

**Pull request merged**